File: 0342-power-of-four/0342-power-of-four.cs
public class Solution {
    public bool IsPowerOfFour(int n) {
        
        long cur = 1;
        
        while (cur <= n && cur < int.MaxValue){
            
            if (cur == n) return true;
            
            cur *= 4;
            
        }
        return false;
    }
}

File: 0342-power-of-four/0342-power-of-four.cpp
class Solution {
public:
    bool isPowerOfFour(int n) {
        if (n == 1) return 1;
        
        if (n & 1) return false;
        
        long long cur = 1;
        
        while (cur < n && cur <= INT_MAX){
            cur *= 4;
            if (cur == n) return true;
            if (cur > n) return false;
        }
        return false;
        
    }
};